addons/mail/static/src/js/systray/systray_messaging_menu.js

```diff
@@ -315,8 +315,16 @@ var MessagingMenu = Widget.extend({
      */
     _onClickPreviewMarkAsRead: function (ev) {
         ev.stopPropagation();
-        var thread;
         var $preview = $(ev.currentTarget).closest('.o_mail_preview');
+        this._processPreviewMarkAsRead($preview);
+    },
+    /**
+     * Process Preview Mark As Read
```

Comment:
Could  you explain a bit more? For example, what happens in case of 'mail_failure'

Reply:
Could you explain what is your motivation to extract it in another function? (commit description)